src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,590 @@
+                + Arrays.toString(anomalyDetector.getIndices().toArray(new String[0]));
+            failuresMap.computeIfAbsent("others", k -> new ArrayList<>()).add(errorMsg);
+            validateAnomalyDetectorResponse();
+        } else {
+            checkADNameExists(detectorId);
+        }
+    }
+
+    private void checkADNameExists(String detectorId) throws IOException {
+        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+        boolQueryBuilder.must(QueryBuilders.termQuery("name.keyword", anomalyDetector.getName()));
+        if (StringUtils.isNotBlank(detectorId)) {
+            boolQueryBuilder.mustNot(QueryBuilders.termQuery(RestHandlerUtils._ID, detectorId));
+        }
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().query(boolQueryBuilder).timeout(requestTimeout);
+        SearchRequest searchRequest = new SearchRequest(ANOMALY_DETECTORS_INDEX).source(searchSourceBuilder);
+        client
+            .search(
+                searchRequest,
+                ActionListener
+                    .wrap(
+                        searchResponse -> onSearchADNameResponse(searchResponse, anomalyDetector.getName()),
+                        exception -> onFailure(exception)
+                    )
+            );
+    }
+
+    private void onSearchADNameResponse(SearchResponse response, String name) throws IOException {
+        if (response.getHits().getTotalHits().value > 0) {
+            failuresMap.computeIfAbsent("duplicates", k -> new ArrayList<>()).add(name);
+            validateAnomalyDetectorResponse();
+        } else {
+            checkForHistoricalData();
+        }
+    }
+
+    public void checkForHistoricalData() {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder()
+            .aggregation(AggregationBuilders.max(AGG_NAME_MAX).field(anomalyDetector.getTimeField()))
+            .size(1)
+            .sort(new FieldSortBuilder(anomalyDetector.getTimeField()).order(SortOrder.DESC));
+        SearchRequest searchRequest = new SearchRequest().indices(anomalyDetector.getIndices().get(0)).source(searchSourceBuilder);
+        client
+            .search(
+                searchRequest,
+                ActionListener
+                    .wrap(response -> checkIfAnyHistoricalData(getLatestDataTime(response)), exception -> { onFailure(exception); })
+            );
+    }
+
+    private void checkIfAnyHistoricalData(Optional<Long> lastTimeStamp) {
+        if (lastTimeStamp.isPresent() && (Instant.now().toEpochMilli() - HISTORICAL_CHECK_IN_MS > lastTimeStamp.get())) {
+            failuresMap.computeIfAbsent("others", k -> new ArrayList<>()).add("No historical data for past 3 months");
+            validateAnomalyDetectorResponse();
+        } else {
+            queryFilterValidation();
+        }
+    }
+
+    private long[] startEndTimeRangeWithIntervals(int numOfIntervals) {
```

Comment:
I would prefer having a `DateTimeRange` class with `start`, `end` attribute

Reply:
I didn't end up implementing this since I changed the method to be even shorted so it seemed weird to create a whole class when I created the start and end time with 4 lines of code now and not used outside my api. In other locations in the code when a similar concept of range was needed, a DateTimeRange didn't seem to be implemented

Reply:
I think such DateTimeRange class should be created which provides better readability and can be re-usable for future code change. I can be possible that in the future the following contributor may handle similar case involving data range

Reply:
Added class